packages/kinvey-datastore2/test/datastore.spec.js

```diff
@@ -0,0 +1,79 @@
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import { DataStore, DataStoreType, collection } from '../src';
+
+describe('DataStoreType', () => {
+  it('should have type Cache', () => {
+    expect(DataStoreType.Cache).to.not.be.undefined;
+  });
+
+  it('should have type Network', () => {
+    expect(DataStoreType.Network).to.not.be.undefined;
+  });
+
+  it('should have type Sync', () => {
+    expect(DataStoreType.Sync).to.not.be.undefined;
+  });
+});
+
+describe('DataStore', () => {
+  describe('constructor', () => {
+    it('should throw and error if a collection is not defined', () => {
+      expect(() => new DataStore()).to.throw();
+    });
+
+    it('should set the collection', () => {
+      const collectionName = 'test';
+      const datastore = new DataStore(collectionName);
+      expect(datastore.collection).to.equal(collectionName);
+    });
+  });
+
+  describe('save()', () => {
+    let datastore;
+
+    before(() => {
+      datastore = new DataStore('test');
+    });
+
+    it('should call create()', () => {
+      const createSpy = spy(datastore, 'create');
+      const entity = {};
+      const options = {};
+      datastore.save(entity, options);
+      expect(createSpy.called).to.be.true;
+      expect(createSpy.calledWith(entity, options)).to.be.true;
+    });
+  });
+
+  describe('create()', () => {
+    let datastore;
+
+    before(() => {
+      datastore = new DataStore('test');
+    });
+
+    it('should create an entity');
+  });
+});
+
+describe('collection()', () => {
+  it('should throw and error if a collection is not defined', () => {
+    expect(() => collection()).to.throw();
+  });
+
+  it('should return a DataStore instance', () => {
+    const datastore = collection('');
+    expect(datastore).to.be.an.instanceof(DataStore);
+  });
+
+  it('should set the collection', () => {
+    const collectionName = 'test';
+    const datastore = collection(collectionName);
+    expect(datastore.collection).to.equal(collectionName);
```

Comment:
I think this is the only test we should have here - there is no logic in `collection()`, other than creating a DataStore instance.

Also having a `collection()` function, not tied to anything is weird - I think we should have it as part of a factory, or reconsider what makes the most sense for it. A function all by itself seems strange.

Reply:
Right now `collection()` only returns a DataStore instance but we will be adding more functionality to it in future PRs. This test will change accordingly.

It isn't all the strange in JavaScript to have a function by itself.

Reply:
It's odd calling collection() and getting a DataStore instance. For internal use, we should call it something else. And I feel `dataStoreFactory.create()` is more approprate. And when we expose it publicly, we can call it collection. I dunno, perhaps we can decide better later on.